src/test/test-memfd-util.c

```diff
@@ -11,21 +9,26 @@
 
 TEST(memfd_get_sealed) {
 #define TEST_TEXT "this is some random test text we are going to write to a memfd"
-        _cleanup_close_ int fd = -EBADF;
-
-        fd = memfd_new_full("test-memfd-get-sealed", MFD_ALLOW_SEALING);
+        _cleanup_close_ int fd = memfd_new_full("test-memfd-get-sealed", MFD_ALLOW_SEALING);
         if (fd < 0) {
-                assert_se(ERRNO_IS_NOT_SUPPORTED(fd));
-                return;
+                ASSERT_TRUE(ERRNO_IS_NOT_SUPPORTED(fd));
+                return (void) log_tests_skipped_errno(fd, "Failed to create new memfd");
         }
-
-        assert_se(write(fd, TEST_TEXT, strlen(TEST_TEXT)) == strlen(TEST_TEXT));
+        ASSERT_OK_EQ_ERRNO(write(fd, TEST_TEXT, strlen(TEST_TEXT)), (ssize_t) strlen(TEST_TEXT));
         /* we'll leave the read offset at the end of the memfd, the fdopen_independent() descriptors should
          * start at the beginning anyway */
 
-        assert_se(memfd_get_sealed(fd) == 0);
-        assert_se(memfd_set_sealed(fd) >= 0);
-        assert_se(memfd_get_sealed(fd) > 0);
+        uint64_t size = 137;
+        ASSERT_OK(memfd_get_size(fd, &size));
+        size = 312;
+        ASSERT_OK(memfd_set_size(fd, size));
+        size = 73;
+        ASSERT_OK(memfd_get_size(fd, &size));
+        size = 31;
+        ASSERT_OK(memfd_set_size(fd, size));
+
+        ASSERT_OK(memfd_set_sealed(fd) >= 0);
```

Comment:
Please add one more call of memfd_get_sealed() _before_ sealing and check if that returns zero, like the original code does. Please do not kill existing test code.